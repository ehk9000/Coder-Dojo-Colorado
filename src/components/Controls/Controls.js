import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Splash from '../Splash/Splash';
import LoginForm from '../LoginForm/LoginForm';
import Profile from '../Profile/Profile';
import CurrentStudents from '../CurrentStudents/CurrentStudents';
import ContactInfo from '../ContactInfo/ContactInfo';
import Family from '../Family/Family';
import RegisterForm from '../RegisterForm/RegisterForm';
import VenuesContainer from '../VenuesContainer/VenuesContainer';
import ChildForm from '../ChildForm/ChildForm';
import EventsContainer from '../EventsContainer/EventsContainer';
import AdminDash from '../AdminDash/AdminDash';

function Controls() {
  return (
    <div>
      <Switch>
        <Route exact path="/" component={Splash} />
        <Route exact path="/venues" component={VenuesContainer} />
        <Route exact path="/venues/:id/events" component={EventsContainer} />
        <Route exact path="/login" component={LoginForm} />
        <Route exact path="/student/:id" component={Profile} />
        <Route exact path="/dashboard/student" component={Profile} />
        <Route exact path="/dashboard/admin" component={AdminDash} />
        <Route exact path="/students" component={CurrentStudents} />
        <Route exact path="/contact/:id" component={ContactInfo} />
        <Route exact path="/myfamily" component={Family} />
        <Route exact path="/myfamily/registerstudent" component={ChildForm} />
        <Route exact path="/dashboard/student" component={Profile} />
        <Route exact path="/dashboard/admin" component={Profile} />
        <Route exact path="/register" component={RegisterForm} />
        <Route render={() => <h1>404 Page Not Found</h1>} />
      </Switch>
    </div>
  );
}

export default Controls;
