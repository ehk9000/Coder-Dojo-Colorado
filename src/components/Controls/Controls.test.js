import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';

describe('Controls', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Controls />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
