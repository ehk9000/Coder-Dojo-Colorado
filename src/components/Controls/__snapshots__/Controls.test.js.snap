// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match snapshot 1`] = `
<div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(VenuesContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/venues"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(EventsContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/venues/:id/events"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(LoginForm)",
          "type": [Function],
        }
      }
      exact={true}
      path="/login"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/student/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/dashboard/student"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/dashboard/admin"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/students"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/contact/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/myfamily"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(RegisterForm)",
          "type": [Function],
        }
      }
      exact={true}
      path="/register"
    />
    <Route
      render={[Function]}
    />
  </Switch>
</div>
`;
